src/main/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptor.java

```diff
@@ -88,6 +88,19 @@ public FormValidation doCheckInstanceCap(
         return FormValidation.validatePositiveInteger(value);
     }
 
+    @Restricted(DoNotUse.class)
+    public FormValidation doCheckInstancesMin(
```

Comment:
This can be done by client side validation: https://github.com/jenkinsci/jenkins/blob/3783cc40a4851d3d92df28f534590b850fe714b2/core/src/main/resources/lib/form/number.jelly#L50

Reply:
It seems the only available client-side validation is either "positive-number" or "required-non-negative-number"; here I want "(non-required) non-negative-number" so I didn't add any client-side validation.

Reply:
oh, nevermind